runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -111,12 +140,23 @@ public void write(IToken token, Object data) throws OverwriteException {
             // Set the data to use the token
             ld.useToken(token);
 
+
             // Do the write
-            l.getReplicationMode(token.getTokenValue())
+            try {
+                l.getReplicationMode(token.getTokenValue())
                         .getReplicationProtocol(runtime)
                         .write(l, ld);
+            } catch (RuntimeException re) {
+                // These two exceptions are pass through. Both of them are already too late for trying
+                // to validate the state of the write, we know that it didn't went through.
+                if (re instanceof SystemUnavailableException || re instanceof OverwriteException) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace the usage of the "instanceof" operator by a catch block. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1193)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace the usage of the "instanceof" operator by a catch block. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1193)